iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOConfig.java

```diff
@@ -62,4 +64,23 @@ FileIO provideLocalFileIO() {
       return null;
     }
   }
+
+  /**
+   * Provides the S3FileIO bean for S3 storage type
+   *
+   * @return S3FileIO bean for S3 storage type, or null if S3 storage type is not configured
+   */
+  @Bean("S3FileIO")
+  S3FileIO provideS3FileIO() {
+    try {
+      S3Client s3 =
+          (S3Client) storageManager.getStorage(StorageType.S3).getClient().getNativeClient();
+      return new S3FileIO(() -> s3);
+    } catch (IllegalArgumentException e) {
+      // If the S3 storage type is not configured, return null
+      // Spring doesn't define the bean if the return value is null
+      log.debug("S3 storage type is not configured", e);
```

Comment:
Should we check here if S3 config is provided and it is illegal then we throw exception?
If it is not provided in the config then makes sense to return null?

Reply:
Done